### PR TITLE
fix: traverse with same ts

### LINKED ISSUE #1696: `Traverse` may get duplicate data if there are same ts records on one pk
* `Traverse` may get duplicate data if there are same ts records on one pk
*  `Traverse` may run in endless loop if there are many same ts records on one pk

### PR DESCRIPTION
**Pull request by dl239**
close #1696

### CODE REVIEW

**Review by vagetablechicken — APPROVED** [review]

File: src/catalog/distribute_iterator.cc [hunk 1]
```diff
@@ -302,19 +302,22 @@ void DistributeWindowIterator::Next() {
         std::string cur_pk = kv_it_->GetPK();
         auto traverse_it = std::dynamic_pointer_cast<openmldb::base::TraverseKvIterator>(kv_it_);
         uint64_t last_ts = 0;
+        uint32_t ts_pos = 1;
```
Comment (zhanghaohit) [inline src/catalog/distribute_iterator.cc:305]: should be 0 or 1 if `traverse_it` is not valid?

File: src/proto/tablet.proto [hunk 2]
```diff
@@ -257,6 +257,7 @@ message TraverseRequest {
     optional uint64 ts = 6;
     optional bool enable_remove_duplicated_record = 7 [default = false];
     optional bool skip_current_pk = 8 [default = false];
+    optional uint64 ts_pos = 9;
```
Comment (zhanghaohit) [inline src/proto/tablet.proto]: `uint32_t`?

File: src/catalog/distribute_iterator.cc (hunk 1 above)
Comment (dl239) [reply src/catalog/distribute_iterator.cc:305]: should be 0 if traverse from another tablet

File: src/proto/tablet.proto [hunk 3]
```diff
@@ -268,6 +269,7 @@ message TraverseResponse {
     optional uint64 ts = 6;
     optional bool is_finish = 7;
     optional uint64 snapshot_id = 8;
+    optional uint64 ts_pos = 9;
```
Comment (zhanghaohit) [inline src/proto/tablet.proto]: same here. `uint32`

**Review by zhanghaohit — APPROVED** [review]
LGTM

**Pull request merged**